### PR TITLE
Fix overlay of page navigation buttons

### LINKED ISSUE #602: Buttons in pagination float over form elements
**Describe the bug**
There is no padding between the Questionnaire items and the next/previous controls in pagination

**Component**
SDC Library

**To Reproduce**
1. Checkout fh/patient-registration-questionnaire and run the SDC Gallery App
2. Click on 'New patient registration' option (last one)
3. Scroll down to the bottom of the first page

**Expected behavior**
Should be padding between the form elements and the next/prev buttons

**Screenshots**
image

**Smartphone (please complete the following information):**
 - Device: virtual device (pixel)
 - Android version: 29

### PR DESCRIPTION
**Pull request by joiskash**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #602 

**Description**
Added a constraint layout to contain the RecyclerView and Page navigation buttons.

**Alternative(s) considered**
No

**Type**
Choose one: Bug fix

**Screenshots (if applicable)**
floating_buttons
not_floating

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

File: datacapture/src/main/res/layout/questionnaire_fragment.xml [hunk 1]
```diff
@@ -18,35 +18,47 @@
...
     android:layout_height="wrap_content"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     android:orientation="vertical"
 >
 
-  <androidx.recyclerview.widget.RecyclerView
-        android:id="@+id/recycler_view"
+  <androidx.constraintlayout.widget.ConstraintLayout
         android:layout_width="match_parent"
-        android:layout_height="wrap_content"
```
Comment (jingtang10) [inline datacapture/src/main/res/layout/questionnaire_fragment.xml:27]: have you tried if setting this to `0dp` will fix the issue without more changes to the layout?

File: datacapture/src/main/res/layout/questionnaire_fragment.xml [hunk 2]
```diff
@@ -18,35 +18,47 @@
     xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     android:orientation="vertical"
 >
 
-  <androidx.recyclerview.widget.RecyclerView
-        android:id="@+id/recycler_view"
+  <androidx.constraintlayout.widget.ConstraintLayout
```
Comment (jingtang10) [inline datacapture/src/main/res/layout/questionnaire_fragment.xml:25]: if you're adding a `ConstraintLayout`, you can probably drop the parent `RelativeLayout`?

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/main/res/layout/questionnaire_fragment.xml (hunk 1 above)
Comment (jingtang10) [reply datacapture/src/main/res/layout/questionnaire_fragment.xml:27]: https://developer.android.com/guide/topics/ui/layout/relative relative layout should really work with this case.

Comment (joiskash) [reply datacapture/src/main/res/layout/questionnaire_fragment.xml:27]: Yes I have tried this.. It makes the RecyclerView disappear

File: datacapture/src/main/res/layout/questionnaire_fragment.xml (hunk 2 above)
Comment (joiskash) [reply datacapture/src/main/res/layout/questionnaire_fragment.xml:25]: That is true.. I was just not sure.

Comment (jingtang10) [reply datacapture/src/main/res/layout/questionnaire_fragment.xml:25]: lmk when this is done

Comment (joiskash) [reply datacapture/src/main/res/layout/questionnaire_fragment.xml]: done

**Review by jingtang10 — APPROVED** [review]
Thanks!

**Pull request merged**